### ISSUE TITLE
Setting instance_count = 0 leads to error with Terrafrom 0.12

### ISSUE THREAD

**Issue by merlineus** [terraform-aws-modules/terraform-aws-ec2-instance#104]
In my setup I have modules with **instance_count = 0** for some environments.
Upgrading to **Terraform 0.12** breaks this setup:

```
Error: Error in function call

  on .terraform/modules/file_rating_mongo_4/terraform-aws-modules-terraform-aws-ec2-instance-6557b76/outputs.tf line 10, in locals:
  10:   this_placement_group              = compact(concat(coalescelist(aws_instance.this.*.placement_group, aws_instance.this_t2.*.placement_group), [""]))
    |----------------
    | aws_instance.this is empty tuple
    | aws_instance.this_t2 is empty tuple

Call to function "coalescelist" failed: no non-null arguments.
```

This happens because behaviour of **coalescelist** function used in **outputs.tf** has been changed in Terraform 0.12: https://github.com/hashicorp/terraform/issues/21370

**Comment by hieuhtr** [comment]
Hmm, this issue is unpredictable, as far as I know.
I'm facing the same issue when setting `instance_count = "1"` or `instance_count = 1`

I don't have any workaround here, but after some `terraform plan`, this issue is bypassed 
You may try to not define `instance_count` in your module section.

**Comment by butson** [comment]
I don't understand why the setting of thisi_placement_group is different than the other variables.

Shouldn't 

  this_placement_group              = compact(concat(coalescelist(aws_instance.this.*.placement_group, aws_instance.this_t2.*.placement_group), [""]))

be 

  this_placement_group              = compact(coalescelist(aws_instance.this.*.placement_group, aws_instance.this_t2.*.placement_grou, [""]))

or something similiar, that is add an a list of one empty string if others are null

**Comment by antonbabenko** [comment]
Thanks for opening this issue. This has been just fixed and [v2.5.0](https://github.com/terraform-aws-modules/terraform-aws-ec2-instance/releases/v2.5.0) has been released.
